src/components/include/utils/threads/thread_delegate.h

```diff
@@ -45,7 +48,12 @@ class Thread;
  * Thread procedure interface.
  * Look for "threads/thread.h" for example
  */
+#ifdef QT_PORT
+class ThreadDelegate : public QObject {
```

Comment:
Could we avoid ``QObject`` inheritance?
As I see ``close_thread`` signal [DirectConnection connected](http://doc.qt.io/qt-5/qt.html#ConnectionType-enum)  and could be replaced with common method call.